### PR TITLE
Remove unused ReaderMethodCache and  WriterMethodCache  constants from ActiveRecord

### PR DESCRIPTION
**Pull request by PareshGupta**
`Module.methods_transplantable?` is deprecated and waiting for removal. Please refer [here](https://github.com/rails/rails/commit/b2bb2a46)

`AttributeMethodCache` class is also not used anymore. Check [here](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods.rb#L37). We should remove this as well. Shall I add the change in same PR?

### CODE REVIEW

**Comment by rafaelfranca** [comment]
> Shall I add the change in same PR?

Yes, please

File: activerecord/lib/active_record/attribute_methods/read.rb [hunk 1]
```diff
@@ -1,8 +1,12 @@
 module ActiveRecord
   module AttributeMethods
     module Read
-      ReaderMethodCache = Class.new(AttributeMethodCache) {
-        private
+
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/attribute_methods/read.rb]: :scissors: this blank line

**Comment by PareshGupta** [comment]
Unused AttributeMethodCache class is also been removed .

**Comment by PareshGupta** [comment]
@rafaelfranca - any update on this?

**Pull request merged**